### PR TITLE
Handel Error

### PR DESCRIPTION
**Pull request by ranasobeid95**
Relates #2

### CODE REVIEW

**Review by SajedaIsmail — DISMISSED** [review]

**Review by MohammadAlhalaq — DISMISSED** [review]

File: client/src/components/App/index.js [hunk 1]
```diff
@@ -102,14 +122,24 @@ class App extends Component {
               <Route
                 exact
                 path="/students/:subjectId/activites/:classId"
-                render={props => <Activities {...props} />}
+                component={Activities}
```
Comment (aminalakhsham) [inline client/src/components/App/index.js]: why did you use component

File: client/src/components/pages/SignUpParent/index.js [hunk 2]
```diff
@@ -31,24 +33,26 @@ export default class SignUp extends React.Component {
...
-        return this.setState({ errors: objError });
+        if (error.toString().includes('ValidationError')) {
+          const objError = {};
+          error.inner.forEach(fielderror => {
+            objError[fielderror.path] = fielderror.message;
+          });
+          return this.setState({ errors: objError });
+        }
+        history.push('/serverError');
+        return '';
```
Comment (aminalakhsham) [inline client/src/components/pages/SignUpParent/index.js:55]:
```suggestion
        return;
```

File: server/controllers/error/serverError.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 const serverError = (err, req, res, next) => {
+  console.log(err);
```
Comment (aminalakhsham) [inline server/controllers/error/serverError.js]: remove console.log

File: server/controllers/routes/parentProfile.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 const { getParentProfile } = require('../../database/quieres/parentProfile');
 
 const parentPRofile = (req, res, next) => {
-  const { id } = req.user;
+  const { id } = req.params;
```
Comment (aminalakhsham) [inline server/controllers/routes/parentProfile.js]: why change this ?!

**Review by aminalakhsham — CHANGES_REQUESTED** [review]

**Review by aminalakhsham — APPROVED** [review]

**Review by MohammadAlhalaq — APPROVED** [review]

**Review by SajedaIsmail — APPROVED** [review]

**Pull request merged**